### PR TITLE
Add constant jerk model simulation for platforms

### PR DESCRIPTION
**Pull request by oharrald-Dstl**
Another method for calculating a series of smooth transitions between a series of target states, using a constant jerk (acceleration time derivative) model.

### CODE REVIEW

**Review by svidal-dstl — APPROVED** [review]

File: docs/examples/Smooth_Platform_Coordinate_Transitions.py [hunk 1]
```diff
@@ -203,3 +210,136 @@
...
+# The next method is :meth:`~simulator.ConstantJerkSimulator.create_models`
+#
+# This method takes a series of states with cartesian kinematic state space elements and returns a
+# chain (list) of :class:`~.ConstantJerkSimulator` transition models and transition times.
+#
+# The :class:`~.ConstantJerkSimulator` type of transition model is an implementation of a
+# noiseless, constant jerk transition in the kinematic, cartesian subspace of the state space.
+#
+# The user will need to provide both initial and final position and velocities. This can be of any
+# dimension. For example, constant jerk in the space :math:`(x, vx, y, vy)` and
```
Comment (jswright-dstl) [inline docs/examples/Smooth_Platform_Coordinate_Transitions.py:227]: I suggest `(x, \dot{x}, y, \dot{y})`.

File: docs/examples/Smooth_Platform_Coordinate_Transitions.py [hunk 2]
```diff
@@ -203,3 +210,136 @@
...
+# :math:`(x, vx, y, vy, z, vz`)` are valid (elements need not be in this order, as a position
```
Comment (jswright-dstl) [inline docs/examples/Smooth_Platform_Coordinate_Transitions.py:228]: I suggest `(x, \dot{x}, y, \dot{y}, z, \dot{z})`. Also there is an extra ` before the close bracket.

**Review by jswright-dstl — APPROVED** [review]

File: docs/examples/Smooth_Platform_Coordinate_Transitions.py [hunk 3]
```diff
@@ -203,3 +210,136 @@
         color='blue',
         linewidth=3)
 fig
+
+
+# %%
+# Constant Jerk
+# ^^^^^^^^^^^^^
+# The next method is :meth:`~simulator.ConstantJerkSimulator.create_models`
```
Comment (sdhiscocks) [inline docs/examples/Smooth_Platform_Coordinate_Transitions.py:218]: This links appears to be broken in documentation.

File: stonesoup/simulator/transition.py [hunk 4]
```diff
@@ -306,3 +307,253 @@ def function(self, state, time_interval, **kwargs):
...
+class ConstantJerkSimulator(TransitionModel):
+    r"""Constant, noiseless, jerk transition model for cartesian space.
+
+    The state space has no acceleration or jerk elements. I.E. the only kinematic components are
+    position and velocity.
+
+    Solution given by :math:`\vec{\ddddot{x}} = \vec{0}`
+
+    The user will provide an initial and final state, each of which containing initial cartesian
+    position and velocities. For example, :math:`(x, vx, y, vy)`.
```
Comment (jswright-dstl) [inline stonesoup/simulator/transition.py:321]: `(x, \dot{x}, y \dot{y})`.

**Pull request merged**